src/vpoller/worker.py

```diff
@@ -160,6 +162,40 @@ def load_config(self):
             'Worker Manager configuration: %s',
             self.config
         )
+
+
+    def check_workers(self):
+        for index in range(len(self.workers)):
+            worker = self.workers[index]
+            if worker.exitcode:
+                logger.error(
+                    '%s died with signal %s. Restarting...',
+                    worker.name,
+                    worker.exitcode
+                )
+                worker.join(3)
+                time.sleep(1)
+                self.start_worker(index)
+
+
+    def start_worker(self,index):
```

Comment:
Please check above comment and possibly remove `index` argument, but simply pass a `worker` instance when iterating over the `self.workers` list.